sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -290,8 +284,8 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         dataSource.createSink(outputMode),
         outputMode,
-        useTempCheckpointLocation = useTempCheckpointLocation,
-        recoverFromCheckpointLocation = recoverFromCheckpointLocation,
+        useTempCheckpointLocation = true,
```

Comment:
If you want "other sinks" to use temporary checkpoint dir, you could achieve this out of box. Modifying here will change the behavior of the current code and lose the restriction for user who miss the setting of checkpoint location unintentionally.